module/Activity/src/Activity/Service/Activity.php

```diff
@@ -110,12 +111,30 @@ public function createActivity(array $params)
         // Find the creator
         $user = $this->getServiceManager()->get('user_role');
         if ($user === 'guest') {
-            throw new \InvalidArgumentException('Guests can not create activities');
+            throw new NotAllowedException('Guests can not create activities');
         }
-        $params['creator'] = $user;
-
         $activity = new ActivityModel();
-        $activity->create($params);
+        $activity->setBeginTime(new \DateTime($params['beginTime']));
+        $activity->setEndTime(new \DateTime($params['endTime']));
+        $activity->setName($params['name']);
+        $activity->setNameEn($params['name_en']);
+        $activity->setLocation($params['location']);
+        $activity->setLocationEn($params['location_en']);
+
+        if (!$params['costs_unknown']) {
+            $activity->setCosts($params['costs']);
+            $activity->setCostsEn($params['costs_en']);
+        }
```

Comment:
There is till some conversions that are to be done, so it would not save that many lines of code.

The form object should only be used in the controller, since it mostly defines the form that a used sees. The data that is used there may or may not correspond to the data you want to set in your model. Therefore the form object should not be used in the model. Furthermore, that allows activities only to be created using forms (or at least implies this). This may not be the case. The only reason that you might use the form in the service because ZF2 put its validation in its form, which is a design flaw IMO.